### PR TITLE
Introduce --xunit-output to emit XUnit-compatible testsuite reports

### PR DESCRIPTION
**Pull request by pmderodat**

### CODE REVIEW

File: e3/testsuite/__init__.py [hunk 1]
```diff
@@ -18,6 +18,11 @@
 import re
 import tempfile
 
+try:
+    from lxml import etree
+except ImportError:
+    import xml.etree.ElementTree as etree
```
Comment (Nikokrock) [inline e3/testsuite/__init__.py]: Any advantage in trying to use lxml here ? If modules from standard library are enough I would just use them

File: e3/testsuite/__init__.py [hunk 2]
```diff
@@ -196,6 +204,12 @@ def testsuite_main(self, args=None):
             " file can then be sourced from a Bourne shell to recreate"
             " the environement that existed when this testsuite was run"
             " to produce a given testsuite report.")
+        parser.add_argument(
+            "--xunit-output",
+            dest="xunit_output",
```
Comment (Nikokrock) [inline e3/testsuite/__init__.py]: you can add metavar="FILE" ?

File: e3/testsuite/__init__.py [hunk 3]
```diff
@@ -321,6 +337,90 @@ def dump_testsuite_result(self):
         """To be implemented."""
         pass
 
+    def dump_xunit_report(self, filename):
```
Comment (Nikokrock) [inline e3/testsuite/__init__.py]: My be a good thing to have by default the possibility to generat xunit reports. Though I would put the implementation in a separate module (e3.testsuite.report.xunit ?)

File: e3/testsuite/__init__.py (hunk 1 above)
Comment (pmderodat) [reply e3/testsuite/__init__.py]: Nothing relevant, I’ll remove it.

File: e3/testsuite/__init__.py (hunk 2 above)
Comment (pmderodat) [reply e3/testsuite/__init__.py]: Sure, will do.

File: e3/testsuite/__init__.py (hunk 3 above)
Comment (pmderodat) [reply e3/testsuite/__init__.py]: Ok

**Comment by pmderodat** [comment]
Oops… Good catch!

By the way, PyYAML emits a warning when running through the call to `yaml.load`. It advises to call `yaml.safe_load` instead, but that does not work, because we rely on the serialization of `TestResult`, which `safe_load` rejects. We don’t seem to have any such problem in the rest of e3 using `yaml.load`: would you by any chance have a recommandation to avoid this warning?

Thank you!

**Pull request merged**